--- xf86-video-geode-2.11.14/src/geode_driver.c~	2012-11-24 10:01:15.000000000 +0000
+++ xf86-video-geode-2.11.14/src/geode_driver.c	2013-03-23 23:21:31.777565077 +0000
@@ -59,8 +59,6 @@
 /* Machine independent stuff */
 #include "mipointer.h"
 #include "micmap.h"
-/* All drivers implementing backing store need this */
-#include "mibstore.h"
 #include "vgaHW.h"
 #include "vbe.h"
 
--- xf86-video-geode-2.11.14/src/gx_driver.c~	2012-11-24 11:01:15.000000000 +0100
+++ xf86-video-geode-2.11.14/src/gx_driver.c	2013-04-06 03:06:45.797019854 +0200
@@ -1350,7 +1350,6 @@ GXScreenInit(SCREEN_INIT_ARGS_DECL)
     if (!pGeode->NoAccel)
         GXAccelInit(pScrn);
 
-    miInitializeBackingStore(pScrn);
     xf86SetBackingStore(pScrn);
 
     /* Set up the soft cursor */
--- xf86-video-geode-2.11.14/src/lx_driver.c~	2012-11-24 11:01:15.000000000 +0100
+++ xf86-video-geode-2.11.14/src/lx_driver.c	2013-04-06 03:06:53.143798310 +0200
@@ -932,7 +932,6 @@ LXScreenInit(SCREEN_INIT_ARGS_DECL)
     if (!pGeode->NoAccel)
         pGeode->NoAccel = LXExaInit(pScrn) ? FALSE : TRUE;
 
-    miInitializeBackingStore(pScrn);
     xf86SetBackingStore(pScrn);
 
     /* Set up the soft cursor */
