diff -p -up xf86-video-geode-2.9.0/src/cim/cim_vg.c.sds xf86-video-geode-2.9.0/src/cim/cim_vg.c
--- xf86-video-geode-2.9.0/src/cim/cim_vg.c.sds	2008-05-07 15:29:06.000000000 -0400
+++ xf86-video-geode-2.9.0/src/cim/cim_vg.c	2008-05-29 09:05:24.000000000 -0400
@@ -430,6 +430,7 @@ vg_set_custom_mode(VG_DISPLAY_MODE * mod
         if (mode_params->src_width != mode_params->mode_width) {
             starting_width = (mode_params->hactive * mode_params->src_width) /
                 mode_params->mode_width;
+            starting_width = mode_params->src_width;
             hscale = (mode_params->src_width << 14) /
                 (mode_params->mode_width - 1);
             irq_ctl |= (DC3_IRQFILT_ALPHA_FILT_EN | DC3_IRQFILT_GFX_FILT_EN);
@@ -440,6 +441,7 @@ vg_set_custom_mode(VG_DISPLAY_MODE * mod
         if (mode_params->src_height != mode_params->mode_height) {
             starting_height = (output_height * mode_params->src_height) /
                 mode_params->mode_height;
+            starting_height = mode_params->src_height;
             vscale = (mode_params->src_height << 14) /
                 (mode_params->mode_height - 1);
             irq_ctl |= (DC3_IRQFILT_ALPHA_FILT_EN | DC3_IRQFILT_GFX_FILT_EN);
@@ -467,11 +469,13 @@ vg_set_custom_mode(VG_DISPLAY_MODE * mod
 
     starting_width = (starting_width + 7) & 0xFFFF8;
 
+    /*
     if (mode_params->hactive < (starting_width >> 1) ||
         output_height < (starting_height >> 1) ||
         (irq_ctl && (starting_width > 1024))) {
         return CIM_STATUS_INVALIDSCALE;
     }
+    */
 
     /* VERIFY INTERLACED SCALING */
     /* The output width must be less than or equal to 1024 pixels when the */
diff -p -up xf86-video-geode-2.9.0/src/lx_driver.c.sds xf86-video-geode-2.9.0/src/lx_driver.c
--- xf86-video-geode-2.9.0/src/lx_driver.c.sds	2008-05-29 09:04:36.000000000 -0400
+++ xf86-video-geode-2.9.0/src/lx_driver.c	2008-05-29 09:04:44.000000000 -0400
@@ -793,7 +793,10 @@ LXPreInit(ScrnInfoPtr pScrni, int flags)
     }
 
     xf86SetCrtcForModes(pScrni, 0);
-    pScrni->currentMode = pScrni->modes;
+    if (pScrni->modes->CrtcVDisplay > 600 && pScrni->modes->next) {
+	pScrni->currentMode = pScrni->modes->next;
+    } else 
+	pScrni->currentMode = pScrni->modes;
 
     xf86PrintModes(pScrni);
     xf86SetDpi(pScrni, 0, 0);
@@ -994,8 +997,15 @@ LXSwitchMode(int index, DisplayModePtr p
     ret = LXAllocateMemory(pScrni->pScreen, pScrni, rotate);
 
     if (ret) {
-	if (pGeode->curMode != pMode)
+	if (pGeode->curMode != pMode) {
+	    if (pMode->CrtcVDisplay > 600) {
+		pGeode->Output = OUTPUT_CRT;
+	    } else {
+		pGeode->Output = OUTPUT_PANEL | OUTPUT_CRT;
+		df_set_crt_enable(DF_CRT_ENABLE);
+	    }
 	    ret = LXSetVideoMode(pScrni, pMode);
+	}
     }
 
     if (ret)
